qutebrowser/browser/dirbrowser.py

```diff
@@ -0,0 +1,68 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Antoni Boucher (antoyo) <bouanto@zoho.com>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The directory browser page."""
+
+import os
+import pkg_resources
+
+import qutebrowser
+from qutebrowser.utils import jinja
+
+
+def dirbrowser(url):
+    """Get the directory browser web page.
+
+    Args:
+        url: The directory path.
+
+    Return:
+        The HTML of the web page.
+    """
+    title = "Browse directory: {}".format(url)
+    template = jinja.env.get_template('dirbrowser.html')
+    # pylint: disable=no-member
+    # https://bitbucket.org/logilab/pylint/issue/490/
+
+    folder = pkg_resources.resource_filename(qutebrowser.__name__,
+                                             'img/folder.png')
+    file = pkg_resources.resource_filename(qutebrowser.__name__,
+                                           'img/file.png')
+
+    def is_file(file):
+        return os.path.isfile(os.path.join(url, file))
+
+    def is_dir(file):
+        return os.path.isdir(os.path.join(url, file))
+
+    if os.path.dirname(url) == url:
+        parent = None
+    else:
+        parent = os.path.dirname(url)
+    all_files = os.listdir(url)
+    files = sorted([{'name': file, 'absname': os.path.join(url, file)}
+                   for file in all_files if is_file(file)],
+                   key=lambda v: v['name'].lower())
+    directories = sorted([{'name': file, 'absname': os.path.join(url, file)}
+                         for file in all_files if is_dir(file)],
+                         key=lambda v: v['name'].lower())
```

Comment:
This looks a bit too cramped IMHO - what about adding a `get_file_list` function or so which looks like this:

```python
def get_file_list(all_files, filter_func):
    items = [{'name': filename, 'absname': os.path.join(url, filename)}
             for filename in all_files if filterfunc(filename)]
    return sorted(items, key=lambda v: v['name'].lower())
```

And then using something like:

```
all_files = os.listdir(url)
files = get_file_list(all_files, is_file)
directories = get_file_list(all_files, is_directory)
```

I'd also appreciate if you could write some tests for `get_file_list`, `is_file` and `is_dir` then, as at some point I'd like to have good test coverage for everything.